magma/interface.py

```diff
@@ -195,7 +188,7 @@ def __getitem__(self, key):
         if isinstance(key, int):
             return self.arguments()[key]
         if isinstance(key, slice):
-            return array([self[i] for i in range(*key.indices(len(self)))])
+            return [self[i] for i in range(*key.indices(len(self)))]
```

Comment:
this seems like a pretty big change in semantics,

Reply:
Hmm, intuitively I would think that slicing an array would return an array

Reply:
To avoid the circular dependency, what about `type(self)(...)`? This seems more extensible and proper than the old implementation too.

Reply:
Actually, I guess you would need to do `type(self).unsized_t[len(slice)](...)` (assuming we have an unsized_t, if not we can add it)

Reply:
this feature allows slicing on an interface* to return a magma `Array` rather than a python array, which could be a handy feature but seems pretty arbitrary as far as the external interface (whether or not it's the caller's responsibility to wrap the returned value in an `Array`).

TBH, this was one of the points I was least sure about since it changes the external interface. However supporting this interface *absolutely* injects a circular dependency since Array -> Circuit -> Interface -> Array. If we must preserve this interface, we can and do a package import. But also note there was only once place in all the repo where this was used (`tests/test_wire/test_arg.py `).

Reply:
Oh sorry, I misread, I thought this was slice of `Array`, yea I'm not sure about this.  Let me consider it, but I don't feel to strongly about either way.

Reply:
@leonardt any final thoughts on this? Don't feel strongly, but given that the flow passed (see below), it seems safe to break this interface, and I think the new interface still makes sense.